src/renderer/components/MapFilter/ReportView/ReportView.js

```diff
@@ -1,247 +1,45 @@
 // @flow
-import React, { useState, useMemo } from 'react'
-import { makeStyles } from '@material-ui/core/styles'
-import { BlobProvider } from '@react-pdf/renderer'
-import deepEqual from 'deep-equal'
-import Button from '@material-ui/core/Button'
-
-import Loading from '../../Loading'
-import CenteredText from '../../CenteredText'
+import React from 'react'
 import ViewWrapper, { type CommonViewProps } from '../ViewWrapper'
-import Toolbar from '../internal/Toolbar'
-import { defineMessages, FormattedMessage, useIntl } from 'react-intl'
-import HideFieldsButton from './HideFieldsButton'
-import { fieldKeyToLabel } from '../utils/strings'
-import getStats from '../stats'
-import PdfViewer from './PdfViewer'
-import PrintButton from './PrintButton'
-import PDFReport from './PDFReport'
-
-import type { Observation } from 'mapeo-schema'
-import type { PresetWithAdditionalFields, FieldState, Field } from '../types'
-import { SettingsContext } from '../internal/Context'
+import { type MapViewContentProps } from '../MapView/MapViewContent'
+import ReportViewContent from './ReportViewContent'
 
 type Props = {
-  ...$Exact<CommonViewProps>
-}
-
-const hiddenTags = {
-  categoryId: true,
-  notes: true,
-  note: true
+  ...$Exact<CommonViewProps>,
+  ...$Exact<MapViewContentProps>
 }
 
-const m = defineMessages({
-  // Displayed whilst observations and presets load
-  noReport: 'No observations available.',
-  nextPage: 'Next',
-  prevPage: 'Previous',
-  previewMessage: 'Previewing first {numPages} pages' // TODO: pluralize
-})
-
 const ReportView = ({
   observations,
   onUpdateObservation,
   onDeleteObservation,
   mapboxAccessToken,
-  bounds,
   mapStyle,
   presets,
   filter,
   getMediaUrl,
   ...otherProps
 }: Props) => {
```

Comment:
The reason I kept this stuff inside this file is that it's difficult to know which file contains which logic just by their names -- ReportView, ReportViewContent, PDFReport, PdfViewer?

Is there a way we can make these names more memorable?

Reply:
Yes I agree (naming things is hard!). I tried to follow this convention for all the views:

`ComponentNameView` e.g. `MediaView` = just wraps the component in `ViewWrapper` which passes the filtered observations and accessor functions, doesn't render anything

`ComponentNameViewContent` e.g. `MediaViewContent` is the logic of the view.

Anything else is sub-components that make sense to move into their own file.

Happy for suggestions for clearer naming.

Reply:
I'd put the ViewWrapper code in `index.js` but since this is the case in other components as well, maybe too much work for what it's worth at this moment.. label it under 'future developer experience' issues